### PR TITLE
C#: Add references to threat modeling to C# Models-as-Data documentation

### PR DESCRIPTION
**Pull request by egregius313**
Adds information about threat modeling to the C# MaD docs.

### CODE REVIEW

**Review by michaelnebel — APPROVED** [review]
Looks good to me, but let's also get a docs review :-)

File: docs/codeql/codeql-language-guides/customizing-library-models-for-csharp.rst [hunk 1]
```diff
@@ -340,3 +340,19 @@ The first four values identify the callable (in this case the getter of the ``No
...
+.. _threat-models-csharp:
+
+Threat models
+-------------
+
+.. include:: ../reusables/beta-note-threat-models.rst
+
+A threat model is a named class of dataflow sources that can be enabled or disabled independently. Threat models allow you to control the set of dataflow sources that you want to consider unsafe. For example, one codebase may only consider remote HTTP requests to be tainted, whereas another may also consider data from local files to be unsafe. You can use threat models to ensure that the relevant taint sources are used in a CodeQL analysis.
+
+The ``kind`` property of the ``sourceModel`` determines which threat model a source is associated with. There are two main categories:
```
Comment (subatoi) [inline docs/codeql/codeql-language-guides/customizing-library-models-for-csharp.rst:353]:
```suggestion
The ``kind`` property of ``sourceModel`` determines which threat model a source is associated with. There are two main categories:
```

Probably debatable but generally I wouldn't expect to see the article before (for example) `sourceModel`. Very minor, in any case!

**Review by subatoi — APPROVED** [review]
This looks good to me docs-wise, thank you! I'm not 100% sure when it should be merged, though, I'm afraid

**Review by michaelnebel — APPROVED** [review]
LGTM!

**Comment by michaelnebel** [comment]
> This looks good to me docs-wise, thank you! I'm not 100% sure when it should be merged, though, I'm afraid

We will align this with product before merging. Thank you!

**Pull request merged**